src/raftstore/store/snap.rs

```diff
@@ -383,14 +721,32 @@ impl SnapManagerCore {
         self.registry.contains_key(key)
     }
 
-    #[inline]
-    pub fn get_snap_file(&self, key: &SnapKey, is_sending: bool) -> io::Result<SnapFile> {
-        SnapFile::new(&self.base, self.snap_size.clone(), is_sending, key)
+    pub fn get_snapshot_to_build(&self, key: &SnapKey) -> io::Result<Box<Snapshot>> {
+        let f = try!(v1::Snap::new_for_writing(&self.base, self.snap_size.clone(), true, key));
+        Ok(Box::new(f))
+    }
+
+    pub fn get_snapshot_for_reading(&self, key: &SnapKey) -> io::Result<Box<Snapshot>> {
```

Comment:
IMO It's a side effect to bind traits to only one. The user could get_snapshot_for_writing and then call apply on it.